synapse/lib/httpapi.py

```diff
@@ -507,14 +571,20 @@ async def get(self):
         if opts is None:
             return
 
-        view = self.cell._viewFromOpts(opts)
+        flushed = False
+        try:
+            view = self.cell._viewFromOpts(opts)
 
-        taskinfo = {'query': query, 'view': view.iden}
-        await self.cell.boss.promote('storm', user=user, info=taskinfo)
+            taskinfo = {'query': query, 'view': view.iden}
+            await self.cell.boss.promote('storm', user=user, info=taskinfo)
 
-        async for pode in view.iterStormPodes(query, opts=opts):
-            self.write(s_json.dumps(pode, newline=jsonlines))
-            await self.flush()
+            async for pode in view.iterStormPodes(query, opts=opts):
+                self.write(s_json.dumps(pode, newline=jsonlines))
+                await self.flush()
+                flushed = True
+        except Exception as e:
```

Comment:
getViewFromOpts could raise an AuthDeny. Should that be handled with a different HTTP response code here?

Reply:
We can easily centralize the mapping for AuthDeny / NoSuchView into a common error handler.